sw/banshee/src/engine.rs

```diff
@@ -326,6 +325,13 @@ impl Engine {
             (0..self.num_clusters).map(|_| tcdm.clone()).collect()
         };
 
+        // Allocate the pripherals
+        let peripherals: Vec<_> = {
+            (0..self.num_clusters)
+                .map(|_| Peripherals::new(&self.config.memory.periphs.callbacks))
+                .collect()
+        };
+
```

Comment:
This is great for now. For Occamy, we will need the same mechanism as for the TCDM, i.e., each cluster has its own peripherals and tcdm, but with different addresses, and the clusters need access to one another's tcdm and peripherals. But that can be tackled in the next step. @paulsc96 Does that also hold for the 'vanilla' Snitch cluster?

Reply:
There is no difference between an Occamy Snitch cluster and a 'vanilla' one except parameterization. So yes, all cluster-local resources should be reachable from anywhere externally and *not* alias with anything else if configured accordingly.

This being said, there could be a usecase for system-wide peripherals. But we should focus on our goal here, which is fully functional multi-cluster emulation.